.github/actions/project_dependencies/action.yml

```
@@ -0,0 +1,55 @@
+name: ddsrouter_dependencies_ubuntu
+description: Common first step for all jobs. Checkout repository, download dependencies and install required packages.
+
+inputs:
+
+  os:
+    description: Specify runs-on machine to download specific artifact
+    required: true
+
+  cmake_build_type:
+    description: Specify cmake_build_type option to download specific artifact
+    required: true
+
+  dependencies_artifact_postfix:
+    description: Specify artifact postfix in case it wants to use a manual one
+    required: false
+    default: _nightly
+
+  target_workspace:
+    description: Specify directory to download dependencies
+    required: false
+    default: ${{ github.workspace }}/install
+
+  # This must be passed as an argument because actions do not access to workflow secrets:
+  # Unrecognized named-value: 'secrets'. https://github.com/orgs/community/discussions/27054
+  # Pass argument {{ secrets.GITHUB_TOKEN }} from workflow
+  secret_token:
+    description: 'Secret token to authenticate the WebRequest so it not get a rate limit error.'
+    required: false
+    default: ''
+
+runs:
+  using: composite
+  steps:
+
+    - name: Install Fast DDS dependencies
+      uses: eProsima/eProsima-CI/multiplatform/install_fastdds_dependencies@v0
+      with:
+        cmake_build_type: ${{ inputs.cmake_build_type }}
+
```

This may fix build issue
```suggestion

  - name: Install doxygen ubuntu
      if: ${{ runner.os }} == "ubuntu"
      uses: eProsima/eProsima-CI/ubuntu/install_apt_packages@v0
      with:
        packages: 'doxygen'
        
-name: Install doxygen windows
      if: ${{ runner.os }} == "windows"
      shell: pwsh
      run: |
          choco install doxygen.install
```